rust/blockstore/src/provider.rs

```diff
@@ -99,13 +100,42 @@ impl BlockfileProvider {
         Ok(())
     }
 
+    // todo: deprecate
```

Comment:
does it make sense to migrate all existing codes to `.create()` and `.fork()` as part of this PR?

Reply:
no, that makes this large PR more unreadable.